bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,570 @@
+    try {
+      PrivateKey privateKey =
+          SecurityUtils.loadPrivateKeyFromKeyStore(
+              SecurityUtils.getPkcs12KeyStore(),
+              new FileInputStream(privateKeyFile),
+              "notasecret",
+              "privatekey",
+              "notasecret");
+
+      return ServiceAccountJwtAccessCredentials.newBuilder()
+          .setClientEmail(serviceAccountEmail)
+          .setPrivateKey(privateKey)
+          .build();
+    } catch (GeneralSecurityException exception) {
+      throw new RuntimeException("exception while retrieving credentials", exception);
+    }
+  }
+
+  /** Creates {@link TransportChannelProvider} for plaintext negotiation type. */
+  private TransportChannelProvider buildPlainTextChannelProvider(String endpoint) {
+
+    InstantiatingGrpcChannelProvider.Builder channelBuilder =
+        defaultGrpcTransportProviderBuilder()
+            .setEndpoint(endpoint)
+            .setChannelConfigurator(
+                new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                  @Override
+                  public ManagedChannelBuilder apply(ManagedChannelBuilder channelBuilder) {
+                    return channelBuilder.usePlaintext();
+                  }
+                });
+
+    String channelCount = configuration.get(BIGTABLE_DATA_CHANNEL_COUNT_KEY);
+    if (!isNullOrEmpty(channelCount)) {
+      channelBuilder.setPoolSize(Integer.parseInt(channelCount));
+    }
+
+    return channelBuilder.build();
+  }
+
+  /** Creates {@link Set} of {@link StatusCode.Code} from {@link Status.Code} */
+  private Set<StatusCode.Code> buildRetryCodes(Set<StatusCode.Code> retryableCodes) {
+    ImmutableSet.Builder<StatusCode.Code> statusCodeBuilder = ImmutableSet.builder();
+
+    // Disables retries for all data operations
+    if (!configuration.getBoolean(ENABLE_GRPC_RETRIES_KEY, true)) {
+      return statusCodeBuilder.build();
+    }
+
+    statusCodeBuilder.addAll(retryableCodes);
+
+    String retryCodes = configuration.get(ADDITIONAL_RETRY_CODES, "");
+
+    for (String stringCode : retryCodes.split(",")) {
+      String trimmed = stringCode.trim();
+      if (trimmed.isEmpty()) {
+        continue;
+      }
+
+      StatusCode.Code code = StatusCode.Code.valueOf(trimmed);
+      Preconditions.checkNotNull(code, String.format("Unknown status code %s found", stringCode));
+      statusCodeBuilder.add(code);
+      LOG.debug("gRPC retry on: %s", stringCode);
+    }
+
+    return statusCodeBuilder.build();
+  }
+
+  /** Creates {@link RetrySettings} for non-streaming VENEER_ADAPTER method. */
+  private RetrySettings buildIdempotentRetrySettings(RetrySettings originalRetrySettings) {
+    RetrySettings.Builder retryBuilder = originalRetrySettings.toBuilder();
+
+    if (configuration.getBoolean(ALLOW_NO_TIMESTAMP_RETRIES_KEY, false)) {
+      throw new UnsupportedOperationException("Retries without Timestamp is not supported yet.");
```

Comment:
Is this still true? I remember you adding unsafe methods to veneer

Reply:
I did add `RowMuatationEntry#createUnsafe` in the veneer but It is not supported on other mutations, also there is no way to set this property on BigtableDataSettings.

That is why throw UnsupportedOperationException at this moment.

Reply:
ok